### PR TITLE
keep ground() in Effect so older code is still valid

### PR DESCRIPTION
**Pull request by sk7725**
#3184

### CODE REVIEW

File: core/src/mindustry/entities/Effect.java [hunk 1]
```diff
@@ -52,6 +52,17 @@ public Effect layer(float l, float duration){
...
     }
+    
+    public Effect ground(){
+        layer = Layer.debris;
+        return this;
+    }
+    
+    public Effect ground(float duration){
+        layer = Layer.debris;
+        this.layerDuration = duration;
```
Comment (summetdev) [inline core/src/mindustry/entities/Effect.java:63]:
```suggestion
        layerDuration = duration;
```

Comment (sk7725) [reply core/src/mindustry/entities/Effect.java:63]: I don't know, the original code before *and* after #3184 has the "this.".

Comment (summetdev) [reply core/src/mindustry/entities/Effect.java:63]: There is to difference between them, just without `this` is shorter.

Comment (MEEPofFaith) [reply core/src/mindustry/entities/Effect.java:63]: `this` is used where the input value and the variable are called the same thing.
If it were `ground(float layerDuration)`, then you would have to do `this.layerDuration = layerDuration` for it to work properly.
I took a 2 week basic Java camp during the summer.

Comment (L0615T1C5-216AC-9437) [reply core/src/mindustry/entities/Effect.java:63]: since input `duration` is not named `layerDuration` the `this.` is redundant

Comment (sk7725) [reply core/src/mindustry/entities/Effect.java:63]: then what about the functions right above it? the layer() of line 45.

**Comment by MEEPofFaith** [comment]
You do realize that in `js` mod scripting you don't even use `ground()`...
`effect.ground()` turns up invalid and instead you just did `effect.ground = true`

Instead of doing `effect.layer(float)` you just do `effect.layer = float`...

**Comment by sk7725** [comment]
h

**Pull request closed without merging**